processor/configuration/src/main/java/com/redhat/service/smartevents/processor/sources/SourceConfigurator.java

```diff
@@ -0,0 +1,37 @@
+package com.redhat.service.smartevents.processor.sources;
+
+import java.util.Optional;
+
+import com.redhat.service.smartevents.infra.exceptions.definitions.user.ActionProviderException;
+
+public interface SourceConfigurator {
+
+    /**
+     * Get validator bean for specific source type. Required for every source.
+     *
+     * @param sourceType desired source type
+     * @return the validator bean
+     * @throws ActionProviderException if bean is not found
+     */
+    SourceValidator getValidator(String sourceType);
+
+    /**
+     * Get resolver bean for specific source type.
+     * This bean is optional and must be implemented only for non-invokable sources
+     * that needs to be resolved to an invokable sources to be executed.
+     *
+     * @param sourceType desired source type
+     * @return {@link Optional} containing the bean if present, empty otherwise.
+     */
+    Optional<SourceResolver> getResolver(String sourceType);
```

Comment:
Not sure this should be `Optional` either. We must provide a fully configured `Input` action for each Source

Reply:
This class is actually not used because I merged it in `GatewayConfigurator`, where this bean is not `Optional` anymore. I will remove this class.